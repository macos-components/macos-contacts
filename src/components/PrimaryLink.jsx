import React from 'react';
import PropTypes from 'prop-types';

import colors from 'style/colors';

const PrimaryLink = (props) => (
  <button
    style={{
      color: colors.primary,
      cursor: 'pointer',
      border: 'none',
      ...props.style
    }}
    onClick={props.onClick}
    onKeyDown={props.onClick}
  >
    {props.label}
  </button>
);

PrimaryLink.propTypes = {
  style: PropTypes.shape({}),
  onClick: PropTypes.func,
  label: PropTypes.string
};

PrimaryLink.defaultProps = {
  style: {},
  onClick: null,
  label: ''
};

export default PrimaryLink;
