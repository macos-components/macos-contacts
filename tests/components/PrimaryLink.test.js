import React from 'react';
import PrimaryLink from 'components/PrimaryLink';
import renderer from 'react-test-renderer';

test('Renders => components/PrimaryLink', () => {
  const component = renderer.create(
    <PrimaryLink />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
