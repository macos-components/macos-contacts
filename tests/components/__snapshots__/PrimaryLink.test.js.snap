// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders => components/PrimaryLink 1`] = `
<button
  onClick={null}
  onKeyDown={null}
  style={
    Object {
      "border": "none",
      "color": "#3e96fd",
      "cursor": "pointer",
    }
  }
>
  
</button>
`;
